### ISSUE TITLE
[BUG] Can't add record to RemoteFeedbackDataset if it has a suggestion

### ISSUE THREAD

**Issue by tomaarsen** [recognai/rubrix#3790]
**Describe the bug**
If I want to add a record to an already existing FeedbackDataset, and the record has a suggestion, then it will throw an exception:

**Stacktrace and Code to create the bug**
```python

import argilla as rg

rg.init(api_url="https://demo.argilla.io", api_key="...", workspace="demo")

inputs = "What are Aliens?"
outputs = "👾"
dataset = rg.FeedbackDataset.from_argilla("emoji-dataset-prompt")
record = rg.FeedbackRecord(fields={"prompt": inputs}, suggestions=[{"question_name": "response", "value": outputs}])
dataset.add_records(record)
```
```bash
C:\code\argilla\src\argilla\client\client.py:182: UserWarning: You're connecting to Argilla Server 1.16.0 using a different client version (1.17.0-dev).
This may lead to potential compatibility issues during your experience.
To ensure a seamless and optimized connection, we highly recommend aligning your client version with the server version.
  warnings.warn(
Pushing records to Argilla...:   0%|                                                                                                    | 0/1 [00:00<?, ?it/s]
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ c:/code/argilla/demo_cant_add_record.py:10 in <module>                                           │
│                                                                                                  │
│    7 outputs = "👾"                                                                              │
│    8 dataset = rg.FeedbackDataset.from_argilla("emoji-dataset-prompt")                           │
│    9 record = rg.FeedbackRecord(fields={"prompt": inputs}, suggestions=[{"question_name": "re    │
│ ❱ 10 dataset.add_records(record)                                                                 │
│   11                                                                                             │
│                                                                                                  │
│ C:\code\argilla\src\argilla\client\feedback\dataset\remote\base.py:257 in add_records            │
│                                                                                                  │
│   254 │   │   │   │   `FeedbackRecord`, list of dictionaries as a record or dictionary as a      │
│   255 │   │   │   │   record; or if the given records do not match the expected schema.          │
│   256 │   │   """                                                                                │
│ ❱ 257 │   │   self._records.add(records=records, show_progress=show_progress)                    │
│   258 │                                                                                          │
│   259 │   def delete_records(self, records: Union["RemoteFeedbackRecord", List["RemoteFeedback   │
│   260 │   │   """Deletes the given records from the dataset in Argilla.                          │
│                                                                                                  │
│ C:\code\argilla\src\argilla\client\utils.py:61 in wrapper                                        │
│                                                                                                  │
│    58 │   │   │   │   │   f"User with role={user.role} is not allowed to call `{func.__name__}   │
│    59 │   │   │   │   │   f" Only users with role={roles} are allowed to call this function."    │
│    60 │   │   │   │   )                                                                          │
│ ❱  61 │   │   │   return func(*args, **kwargs)                                                   │
│    62 │   │                                                                                      │
│    63 │   │   return wrapper                                                                     │
│    64                                                                                            │
│                                                                                                  │
│ C:\code\argilla\src\argilla\client\feedback\dataset\remote\dataset.py:89 in add                  │
│                                                                                                  │
│    86 │   │   │   for record in records[i : i + PUSHING_BATCH_SIZE]:                             │
│    87 │   │   │   │   if record.suggestions:                                                     │
│    88 │   │   │   │   │   for suggestion in record.suggestions:                                  │
│ ❱  89 │   │   │   │   │   │   suggestion.question_id = self.__question_name2id[suggestion.ques   │
│    90 │   │   │   │   records_batch.append(                                                      │
│    91 │   │   │   │   │   record.dict(exclude={"id": ..., "suggestions": {"__all__": {"questio   │
│    92 │   │   │   │   )                                                                          │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'RemoteFeedbackRecords' object has no attribute '_RemoteFeedbackRecords__question_name2id'
```

The failing line: https://github.com/argilla-io/argilla/blob/97947ede175c400012798166934d08f5d4924621/src/argilla/client/feedback/dataset/remote/dataset.py#L89

**Expected behavior**

**Environment:**
 - Argilla Version [e.g. 1.0.0]: 1.16.0
 - ElasticSearch Version [e.g. 7.10.2]: 8.5.3
 - Docker Image (optional) [e.g. argilla:v1.0.0]: -

**Additional context**
-

**Comment by alvarobartt** [comment]
Hi @tomaarsen thanks for reporting, I'm indeed refactoring the `pydantic.BaseModel` schemas to (hopefully) make those clearer and easier to maintain, see #3784 (what you just reported is already fixed in that branch)

**Comment by alvarobartt** [comment]
Fixed as of #3784 merge, to be released at 1.17.0, thanks for reporting @tomaarsen!

**Comment by tomaarsen** [comment]
Awesome! Thanks
